code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/template/DockerComposeTemplate.xtend

```diff
@@ -0,0 +1,100 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.template
+
+import java.util.Map
+import de.fhdo.lemma.model_processing.code_generation.container_base.file.docker.Service
+import de.fhdo.lemma.model_processing.code_generation.container_base.file.docker.Network
+
+/**
+ * Template class for generating docker compose files.
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class DockerComposeTemplate {
+
+    /*
+     * Build the operation node specific docker compose part
+     */
+    static def String buildDockerComposeParts(String name, Service service) '''
+        «"  "»«name.toLowerCase»:
+        «IF service.image !== null»
+            «"  "»  image: «service.image.toLowerCase»
+        «ENDIF»
+        «IF service.build !== null»
+            «"  "»  build: «service.build.toLowerCase»
+        «ENDIF»
+        «IF service.container_name !== null»
+            «"  "»  container_name: «service.container_name.toLowerCase»
+        «ENDIF»
+        «IF service.environment !== null»
+            «"  "»  environment:
+            «FOR Map.Entry<String, String> environment : service.environment.entrySet»
+                «"  "»    «environment.key»: «environment.value»
```

Comment:
See? ;)

Reply:
same as in [https://github.com/SeelabFhdo/lemma/pull/18#discussion_r519450347](https://github.com/SeelabFhdo/lemma/pull/18#discussion_r519450347)

Reply:
Okay, but here the situation is a bit different. My concern are the empty spaces after the `«"  "»` construct. Actually, it be better then to have something like `«"        "»`.